File: app/src/test/java/com/example/androidfundamentals/MainTestApplication.java
package com.example.androidfundamentals;

import com.example.androidfundamentals.di.AddressComponent;
import com.example.androidfundamentals.di.DaggerAddressComponent;
import com.example.androidfundamentals.di.MockAddressModule;

public class MainTestApplication extends MainApplication {
    @Override
    public void onCreate() {
        //super.onCreate();
    }

    @Override
    protected AddressComponent getComponent() {
        return null;
//                Mok.builder()
//                .addressModule(new MockAddressModule())
//                .build();
    }
}

File: app/src/test/java/com/example/androidfundamentals/MainTestApplication.kt
package com.example.androidfundamentals

import com.example.androidfundamentals.di.DaggerMockApplciationAndroidComponent

class MainTestApplication : MainAndroidApplication() {

    override fun onCreate() {
        component = DaggerMockApplciationAndroidComponent
            .factory()
            .create(applicationContext)

        component.inject(this)
    }
}
